x-pack/plugins/security_solution/public/common/components/search_bar/index.tsx

```diff
@@ -286,24 +282,19 @@ export const SearchBarComponent = memo<SiemSearchBarProps & PropsFromRedux>(
         filterManager.getUpdates$().subscribe({
           next: () => {
             if (isSubscribed) {
-              saveAppStateToStorage(filterManager.getAppFilters());
```

Comment:
There are places where the filterManager is stored in timeline redux currently. It's definitely an anti-pattern we want to move away from, but we don't know how it's all wired through at the moment to do so confidently. I believe the reason is that timeline has its own filter manager and in the event we allow multiple timelines, each will have its own instance of the filterManager. We'd want to track the filterManager state rather than the `filterManager` itself.

Reply:
Ohh, that's nice. I did this here as well https://github.com/elastic/kibana/pull/124513 but we never merged it because we were waiting for the redesign of the filter UI 🎉

Reply:
And to what @michaelolo24 mentioned. Yes, we 100% should remove the instance of filtermanager from Redux. I think @kqualters-elastic found out that this lead to quite the performance impact on some Redux operations. Is there already a ticket for this work?